src/docfx/template/TemplateEngine.cs

```diff
@@ -191,8 +170,61 @@ public JObject PreprocessMetadata(JObject outputMetadata, Document file)
         public JObject TransformTocMetadata(object model)
             => TransformMetadata("toc.json.js", JsonUtility.ToJObject(model));
 
+        private IReadOnlyDictionary<string, (bool isData, bool transformMetadata, (JsonSchemaValidator, JsonSchemaTransformer) jsonSchema)>
+            LoadSchemas(string schemaDir, string contentTemplateDir)
+        {
+            var schemas = new Dictionary<string, (bool isData, bool transformMetadata, (JsonSchemaValidator, JsonSchemaTransformer) jsonSchema)>();
+            var schemaNames = Directory.Exists(schemaDir) ? Directory.EnumerateFiles(schemaDir, "*.schema.json", SearchOption.TopDirectoryOnly)
+                                                    .Select(k => Path.GetFileNameWithoutExtension(k))
+                                                    .Select(k => k.Substring(0, k.Length - ".schema".Length)).ToHashSet() : new HashSet<string>();
+            var allJsFiles = Directory.Exists(contentTemplateDir) ? Directory.EnumerateFiles(contentTemplateDir, "*.js", SearchOption.TopDirectoryOnly)
+                             .Select(k => Path.GetFileNameWithoutExtension(k))
+                             .GroupBy(k => Regex.Replace(k, "(\\.html\\.primary|\\.mta\\.json)$", ""))
+                             .ToDictionary(g => g.Key, g => g.ToList()) : default;
+            foreach (var schemaName in schemaNames)
+            {
+                bool isData = true;
+                bool transformMetadata = false;
+                if (allJsFiles.TryGetValue(schemaName, out var jsFiles))
+                {
+                    foreach (var jsFile in jsFiles)
+                    {
+                        if (jsFile.EndsWith(".html.primary"))
+                            isData = false;
+                        if (jsFile.EndsWith(".mta.json"))
+                            transformMetadata = true;
+                    }
+                }
+                schemas.Add(schemaName, (isData, transformMetadata, new Lazy<(JsonSchemaValidator, JsonSchemaTransformer)>(() => GetJsonSchemaCore(schemaName)).Value));
+            }
+            schemas.Add("LandingData", (false, false, new Lazy<(JsonSchemaValidator, JsonSchemaTransformer)>(() => GetJsonSchemaCore("LandingData")).Value));
+            return schemas;
+
+            (JsonSchemaValidator, JsonSchemaTransformer) GetJsonSchemaCore(string schemaName)
+            {
+                if (schemaName is null)
+                {
+                    return default;
+                }
+
+                var schemaFilePath = Path.Combine(schemaDir, $"{schemaName}.schema.json");
+                if (string.Equals(schemaName, "LandingData", StringComparison.OrdinalIgnoreCase))
+                {
+                    schemaFilePath = Path.Combine(AppContext.BaseDirectory, "data", "schemas", "LandingData.json");
```

Comment:
why is `LandingData` special here not like the others?

Reply:
https://github.com/search?utf8=%E2%9C%93&q=org%3Amicrosoftdocs+landingdata+in%3Afile+extension%3Ayml&ref=simplesearch

LandingData used to belong to Yaml-Pipeline(which I'm not really familiar with), and it doesn't have a corresponding `LandingData.schema.json` defined in template repo